src/session-pool.ts

```diff
@@ -532,17 +532,22 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
   }
 
   /**
-   * Borrows session from specific group.
+   * Borrows the first session from specific group. This method may only be called if the inventory
+   * actually contains a session of the desired type.
    *
    * @private
    *
    * @param {string} type The desired session type.
    * @return {Session}
    */
   _borrowFrom(type: types): Session {
-    const session = this._inventory[type][0];
-    this._borrow(session);
-    return session;
+    const session = this._inventory[type].pop();
```

Comment:
to switch from FIFO to LIFO perhaps we can switch 
from 
```ts
const session = this._inventory[type][0];
```
to 
```ts
const session = this._inventory[type][this._inventory.length - 1];
```
Once [`splice`](https://github.com/googleapis/nodejs-spanner/blob/a1a1ff49ed9036acf0876ae4967ad7f7be2dac27/src/session-pool.ts#L531) gets the last index instead of the first it's efficiency should be same as `pop()`

Reply:
Yes, you're right about that, but in my opinion, using `push` and `pop` better reflects the LIFO behavior of the session pool that we want. This makes it easier for new developers to automatically understand what the behavior is. It also keeps it more in line with the other client libraries.